lib/components/base-components/NormalComponent/NormalComponent.ts

```diff
@@ -150,6 +150,62 @@ export class NormalComponent<
     this.initPorts()
   }
 
+  doInitialSourceNameDuplicateComponentRemoval(): void {
+    // Early return if component has no explicit name (auto-assigned names don't conflict)
+    if (!this.name) return
+
+    const root = this.root!
+
+    // Find all NormalComponent instances with the same name that have already started or completed SourceNameDuplicateComponentRemoval
+    const findEarlierComponentsWithSameName = (
+      component: any,
+      targetName: string,
+    ): any[] => {
+      const matches: any[] = []
+
+      if (
+        component._isNormalComponent &&
+        component.name === targetName &&
+        component !== this &&
+        component.renderPhaseStates.SourceNameDuplicateComponentRemoval
+          .initialized
+      ) {
+        matches.push(component)
+      }
+
+      // Recursively check children
+      if (component.children) {
+        for (const child of component.children) {
+          matches.push(...findEarlierComponentsWithSameName(child, targetName))
+        }
+      }
+
+      return matches
+    }
+
+    const conflictingComponents = findEarlierComponentsWithSameName(
```

Comment:
this feels a bit hacky to me, you should just be able to go `this.getSubcircuit(`.${name}`)` and see if multiple components are returned. If multiple components are returned, then there's an issue. You will still need to check if the component was initialized first, as you're already doing